src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecParametersProvider.cs

```diff
@@ -0,0 +1,99 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+
+using System.Reflection;
+using System.Text.Json;
+using Nethermind.Config;
+
+namespace Nethermind.Specs.ChainSpecStyle;
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Core;
+
+public class ChainSpecParametersProvider : IChainSpecParametersProvider
+{
+    // TODO: test that all IChainSpecEngineParameters have this suffix
+    private const string EngineParamsSuffix = "ChainSpecEngineParameters";
+
+    private readonly Dictionary<string, JsonElement> _chainSpecParameters =
+        new(StringComparer.InvariantCultureIgnoreCase);
+
+    private readonly ConcurrentDictionary<Type, IChainSpecEngineParameters> _instances = new();
```

Comment:
Why does this have to be concurrent? We cannot add elements through `AllChainSpecParameters` and there is no other accessor.